src/main/scala/org/constellation/infrastructure/endpoints/HealthCheckEndpoints.scala

```diff
@@ -0,0 +1,77 @@
+package org.constellation.infrastructure.endpoints
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.syntax._
+import org.constellation.domain.healthcheck.HealthCheckConsensus.{FetchPeerHealthStatus, HealthConsensusCommand, NotifyAboutMissedConsensus, SendPerceivedHealthStatus}
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager.SendProposalError
+import org.constellation.p2p.Cluster.EitherCodec._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.HttpRoutes
+import org.http4s.circe._
+
+class HealthCheckEndpoints[F[_]](implicit F: Concurrent[F]) extends Http4sDsl[F] {
+  Slf4jLogger.getLogger[F]
+
+  def peerHealthCheckEndpoints(healthCheckConsensusManager: HealthCheckConsensusManager[F]) =
+    processConsensusCommand(healthCheckConsensusManager) <+>
+      getRoundProposal(healthCheckConsensusManager) <+>
+      getClusterState(healthCheckConsensusManager)
+
+  private def processConsensusCommand(
+    healthCheckConsensusManager: HealthCheckConsensusManager[F]
+  ): HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "health-check" / "consensus" =>
+      for {
+        cmd <- req.decodeJson[HealthConsensusCommand]
+        response <- cmd match {
+          case SendPerceivedHealthStatus(consensusHealthStatus, asProxyForId) =>
+            {asProxyForId match {
+              case Some(proxyToPeer) =>
+                healthCheckConsensusManager.proxySendHealthProposal(consensusHealthStatus, proxyToPeer)
+              case None =>
+                // are we sure that it will be processes no matter what and therefore we can return Accepted immediately?
```

Comment:
yes, it will work that way. you can take a look into transaction endpoint - it works in the same way. Accepted is being returned and tx is being processed in background

Reply:
That also refers to what is happening in handleConsensusHealthProposal and further where we are modifying some Ref's. And Ref.modify is suspended in F so I guess it can fail, so there is no 100% guarantee that after responding with Accepted the whole flow will not fail. Probably that's not a big issue, because we also have a functionality to fetch missing proposals. Regarding F.start, that's right it works, the healthcheck wouldn't work at all if this bit was failing :).

Reply:
Accepted means it's being processed or queued to being processed, it doesn't mean that it already succeeded. so IMO it's fine